### PR TITLE
X11 path

### PR DESCRIPTION
**Pull request by nieder**
* Apply current PatchScript perl-pie into the code.
* Change code to use /opt/X11/etc/X11/xinit that was implemented with Xquartz 2.8.

Because the lib -> etc change is a direct replacement, we would stop supporting Xquartz < 2.8 (unless the older Xquartz are also installing the `etc/X11/xinit` directory but just not using it).

### CODE REVIEW

**Comment by dmacks** [comment]
These look like simple shell scripts; can they check for /etc/ and then fall back to /lib/ to remain backward-compatible with older XQuartz?

**Comment by nieder** [comment]
How does this look:
```
dobuild() {
    go $mkinstalldirscmd "$builddir"
    $show "creating build.sed"
    if [[ -d /opt/X11/etc/X11/xinit ]]; then
        xinitdir = /opt/X11/etc/X11/xinit;
    elif [[ -d /opt/X11/lib/X11/xinit ]]; then
        xinitdir = /opt/X11/lib/X11/xinit;
    else
    	echo "xinit dir for X11 not found. Exiting."
    	exit 1;
    fi
    case "x$run" in
	x:) ;;
	*)
    $run cat > "$builddir/build.sed" << EOF
s|@PREFIX@|$prefix|g
s|@XINITDIR@|/$xinitdir|g
s|@X_BINDIR@|/opt/X11/bin|g
EOF
	;;
    esac
    for src in "$srcdir"/sedsrc/*.in; do
	base=`basename "$src" .in`
	go_r "$builddir/$base" sed -f "$builddir/build.sed" "$src"
    done
}
```
I added the if..then and substituted then substituted the hardcoded path in build.sed with a variable. Not sure yet how the variable replacement will work when actually applied, but I think it should apply the resolved string and not the variable name.

**Comment by dmacks** [comment]
That looks right.

**Comment by nieder** [comment]
Hmmm. Only one file uses the @XINITDIR@ replacement token transformed by build.sh. The other two files that look for the xinit directory use hardcoded path for it. When using build.sh it hardcodes the xinit location inside the file `sys-xinitrc-fink` to whatever the above conditional resolves. But if we're using Xquartz-2.7 when xinitrc is installed and then update to 2.8, the location will be different (though the `lib` directory will still be present).

Do we know if 2.8 uses anything in X11/lib/X11/init if present?

tl;dr: I guess we should drop the @XINITDIR@ token replacement and use the if...else conditional in all the necessary files?

**Comment by nieder** [comment]
This has somewhat mutated into adding some more changes to the package.
Please review

**Comment by nieder** [comment]
@dmacks ping. I think I've figure out their build system, so would like to get this out

**Comment by nieder** [comment]
@TheSin- approved in #1
